### PR TITLE
Add new package: Mosh to spec-extended

### PR DESCRIPTION
**Pull request by AZaugg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 
Getting more than than a few requests for mariner to have the package mosh added, adding it to specs-extended

###### Change Log  

- New Package addition

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- I was not able to build it locally, was hopping to use a buddy build

### CODE REVIEW

**Review by ChrisCarini — APPROVED** [review]
Excited to see this land! 🎉

**Review by rlmenge — COMMENTED** [review]
Started a buddy build here :) https://dev.azure.com/mariner-org/mariner/_build/results?buildId=561938&view=results

File: SPECS-EXTENDED/mosh/mosh.spec [hunk 1]
```diff
@@ -0,0 +1,99 @@
+Name:		mosh
+Version:	1.4.0
+Release:	1%{?dist}
```
Comment (mfrw) [inline SPECS-EXTENDED/mosh/mosh.spec:3]: A few of the PR checks are failing.

**Comment by rlmenge** [comment]
Unblocked the PR checks and started a new buddy build https://dev.azure.com/mariner-org/mariner/_build/results?buildId=566882&view=results

**Comment by christopherco** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618714&view=results

**Comment by christopherco** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=619765&view=results

File: SPECS-EXTENDED/mosh/mosh.spec [hunk 2]
```diff
@@ -0,0 +1,226 @@
+Name:		mosh
+Version:	1.4.0
+Release:	6%{?dist}
+Summary:	Mobile shell that supports roaming and intelligent local echo
+Vendor:     Microsoft Corporation
+Distribution: Mariner
```
Comment (PawelWMS) [inline SPECS-EXTENDED/mosh/mosh.spec:6]: Could you only replace tabs with spaces?

**Review by PawelWMS — APPROVED** [review]
Looks good, just a small request to use spaces instead of tabs.

**Review by christopherco — APPROVED** [review]

**Pull request merged**